### PR TITLE
machines: Move reusable code to libvirt-common.es6

### PR DESCRIPTION
**Pull request by KKoukiou**
Splitting https://github.com/cockpit-project/cockpit/pull/9240 into smaller pieces to make merge process quicker.

### CODE REVIEW

**Comment by KKoukiou** [comment]
All these functions that I moved can be reused in other provider files.
More fined grained splitting can be done instead of putting everything into libvirt-common file if needed.

File: pkg/machines/components/vmDisksTabLibvirt.jsx [hunk 1]
```diff
@@ -64,7 +64,7 @@ class VmDisksTabLibvirt extends React.Component {
             return null;
         }
 
-        if (vm.status === 'running') {
+        if (vm.state === 'running') {
```
Comment (martinpitt) [inline pkg/machines/components/vmDisksTabLibvirt.jsx]: There's a similar thing here:

    pkg/kubernetes/scripts/virtual-machines/components/VmsListingRow.jsx:    const phase = (vm.status && vm.status.phase) || _("n/a");

but there `vm.status` seems to be an object, while here it's a string. Are these two different `status`es really?

Comment (KKoukiou) [reply pkg/machines/components/vmDisksTabLibvirt.jsx]: I am not familiar with k8s package, I am not sure if it's problem that there as well.

For the commit here we need to check the vm.state string as set here, https://github.com/cockpit-project/cockpit/blob/master/pkg/machines/libvirt.es6#L1048

vm.status is clearly a typo, not initialized anywhere.

Comment (martinpitt) [reply pkg/machines/components/vmDisksTabLibvirt.jsx]: Indeed, thanks! Sorry, I totally missed the /kubernetes, I just saw the "machines" there :)

**Pull request closed without merging**